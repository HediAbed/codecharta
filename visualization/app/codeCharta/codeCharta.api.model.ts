import {
	AttributeTypes,
	AttributeTypeValue,
	CodeMapNode,
	Edge,
	MarkedPackage,
	MetricStatisticsItem,
	RecursivePartial,
	Settings
} from "./codeCharta.model"

export interface ExportCCFile {
	projectName: string
	apiVersion: string
	fileChecksum: string
	nodes: CodeMapNode[]
	attributeTypes?: AttributeTypes | OldAttributeTypes
	edges?: Edge[]
	markedPackages?: MarkedPackage[]
	blacklist?: ExportBlacklistItem[]
	metricStatistics?: ExportMetricStatisticsItem[]
}

export interface ExportBlacklistItem {
	path: string
	type: ExportBlacklistType
}

export type ExportMetricStatisticsItem = MetricStatisticsItem

export enum ExportBlacklistType {
	hide = "hide",
	exclude = "exclude"
}

export enum APIVersions {
	ZERO_POINT_ONE = "0.1",
	ONE_POINT_ZERO = "1.0",
	ONE_POINT_ONE = "1.1",
	ONE_POINT_TWO = "1.2"
}

export interface ExportScenario {
	name: string
	settings: RecursivePartial<Settings>
}

export interface OldAttributeTypes {
	nodes?: [{ [key: string]: AttributeTypeValue }?]
	edges?: [{ [key: string]: AttributeTypeValue }?]
}
