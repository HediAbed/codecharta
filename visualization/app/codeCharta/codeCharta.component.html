<tool-bar-component></tool-bar-component>

<file-extension-bar-component ></file-extension-bar-component>
<div style="margin-top:1px; width: 100%; min-height: 100px; position: fixed; z-index: 10; overflow: hidden; white-space: nowrap;font-size: 0px;" >	
	<ribbon-bar-component class="ribbon-bar-component"></ribbon-bar-component >
</div>


<div id="actionButtons">
	<md-button
		ng-if="$ctrl._viewModel.focusedNodePath.length > 0"
		class="md-fab md-primary dark_shadow_opacity long-button"
		aria-label="Remove Focus on node"
		ng-click="$ctrl.removeFocusedNode()"
	>
		Show complete map
	</md-button>
	<md-button class="md-fab md-primary dark_shadow_opacity" aria-label="Fit Map to View" ng-click="$ctrl.fitMapToView()">
		<i class="fa fa-compress fa-lg"></i>
	</md-button>
</div>

<code-map-component></code-map-component>

<legend-panel-component></legend-panel-component>

<detail-panel-component></detail-panel-component>

<loading-gif-file-component></loading-gif-file-component>

<div id="mw-logo">
	<div class="logo">
		<a href="https://www.maibornwolff.de" target="_blank"><img src="assets/logo.png" alt="MaibornWolff"/></a>
		<h2>
			<span>CodeCharta {{ $ctrl._viewModel.version }}</span>
		</h2>
	</div>
</div>

<node-context-menu-component></node-context-menu-component>
