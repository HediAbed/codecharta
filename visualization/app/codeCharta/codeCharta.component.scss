@import "../mixins.scss";

code-charta-component {
	$animationDuration: 0.5;

	#mw-logo {
		.logo {
			width: 160px;
			height: 100px;
			position: absolute;
			left: 0;
			bottom: 30px;
			z-index: 29;
		}

		a,
		img {
			width: 5em;
			display: block;
			margin: auto !important;
		}

		h2 {
			text-transform: uppercase;
			color: #5a585a;
			font-family: Roboto, sans-serif;
			font-weight: 500;
			font-size: 9pt;
			text-align: center;
			margin: 0;
			z-index: -1;
		}
	}

	#title {
		position: fixed;
		z-index: 2;
		width: 100%;
	}

	.small-info {
		color: rgba(0, 0, 0, 0.38);
		font-size: 9pt;
	}

	#codeMap {
		position: fixed;
		z-index: 0;
		top: 90px;
		left: 0;
		width: 100%;
		height: 100%;
		transition: all $animationDuration + s ease;

		&.expanded {
			height: calc(100% - 170px);
		}
	}

	#actionButtons {
		position: fixed;
		bottom: 2em;
		right: 2em;
		z-index: 60;

		i.fa-folder-open {
			margin: 4px 0 0 2px;
		}

		.long-button {
			width: inherit;
			padding-left: 10px;
			padding-right: 10px;
			border-radius: 100px;
		}

		> * {
			margin-left: 4px;
			margin-right: 4px;
			z-index: 60;
		}
	}

	.dark_shadow_opacity {
		box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2) !important;
		opacity: 0.97;
	}

	.bottom_shadow {
		box-shadow: 0 4px 4px -2px rgba(0, 0, 0, 0.2) !important;
	}

	.md-button.round-button {
		margin-right: 0 !important;
		float: right;
		min-width: 36px;
		width: 36px;
		border-radius: 50%;
	}

	.ribbon-bar-component {
		z-index: 10;

		md-card,
		.section .section-header,
		.section .section-body,
		#toggle-ribbon-bar-fab {
			transition: all $animationDuration + s ease;
		}
	}

	.ribbon-bar-wrapper {
		margin-top: 1px;
		width: 100%;
		min-height: 100px;
		height: 500px;
		position: fixed;
		z-index: 10;
		overflow: hidden;
		white-space: nowrap;
		font-size: 0px;
		pointer-events: none;
	}

	.angular-green {
		color: #00bfa5;
	}

	.small-action-button {
		height: 24px;
		width: 24px;
		float: left;

		*,
		.md-fab:active {
			margin: 0;
			padding: 0;
		}

		.md-fab {
			position: relative;
			min-height: 0;
			width: 100%;
			height: 100%;
			z-index: 0;
			border-radius: 50%;

			i {
				position: absolute;
				top: 5px;
				left: 6px;
				color: white;

				&:hover {
					cursor: pointer;
					color: #eee;
				}
			}
		}
	}
}
