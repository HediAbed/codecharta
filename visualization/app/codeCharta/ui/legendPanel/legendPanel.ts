import angular from "angular";
import "../../state/state.module";
import "../codeMap/codeMap";

import {legendPanelComponent} from "./legendPanel.component";

angular.module("app.codeCharta.ui.legendPanel",["app.codeCharta.state", "app.codeCharta.ui.codeMap"]);

angular.module("app.codeCharta.ui.legendPanel").component(
    legendPanelComponent.selector, legendPanelComponent
);