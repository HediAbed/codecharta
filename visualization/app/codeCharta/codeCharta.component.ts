import { NameDataPair, UrlUtils } from "./util/urlUtils"
import {IHttpService, ILocationService, IRootScopeService} from "angular"
import "./codeCharta.component.scss"
import { NodeContextMenuController } from "./ui/nodeContextMenu/nodeContextMenu.component"
import { CodeMapActionsService } from "./ui/codeMap/codeMap.actions.service"
import { CodeChartaService } from "./codeCharta.service"

/**
 * This is the main controller of the CodeCharta application
 */
export class CodeChartaController {
	public viewModel = {
		version: require("../../package.json").version,
		numberOfLoadingTasks: 0
	}
	private urlUtils: UrlUtils

	/* @ngInject */
	constructor(
		//private threeOrbitControlsService: ThreeOrbitControlsService,
		private $rootScope: IRootScopeService,
		//private dialogService: DialogService,
		//private codeMapActionsService: CodeMapActionsService,
		private codeChartaService: CodeChartaService,
		private $location: ILocationService,
		private $http: IHttpService
	) {
		this.urlUtils = new UrlUtils(this.$location, this.$http)
		this.subscribeToLoadingEvents(this.$rootScope)
		this.loadFileOrSample()
	}

	public fitMapToView() {
		// this.threeOrbitControlsService.autoFitTo()
	}

	public removeFocusedNode() {
		//this.codeMapActionsService.removeFocusedNode()
	}

	public loadFileOrSample() {
		this.viewModel.numberOfLoadingTasks++
		return this.urlUtils.getFileDataFromQueryParam().then((data: NameDataPair[]) => {
			if (data.length > 0) {
				this.tryLoadingFiles(data)
			} else {
				this.tryLoadingSampleFiles()
			}
		}, this.tryLoadingSampleFiles.bind(this))
	}

	public tryLoadingSampleFiles() {
		if (this.urlUtils.getParam("file")) {
			/*this.dialogService.showErrorDialog(
				"One or more files from the given file URL parameter could not be loaded. Loading sample files instead."
			)*/
		}
		this.tryLoadingFiles([
            { name: "sample1.json", data: require("./assets/sample1.json") },
            { name: "sample2.json", data: require("./assets/sample2.json") }
        ]);
    }
    
    private tryLoadingFiles(values: NameDataPair[]) {
        this.codeChartaService
        .loadFiles(
            values
        )
        .then(() => this.viewModel.numberOfLoadingTasks--)
        .catch(e => {
			this.viewModel.numberOfLoadingTasks--
			console.error(e);
            this.printErrors(e)
        })
    }

	public printErrors(errors: Object) {
		//this.dialogService.showErrorDialog(JSON.stringify(errors, null, "\t"))
	}

	private subscribeToLoadingEvents($rootScope: angular.IRootScopeService) {
		$rootScope.$on("add-loading-task", () => {
			this.viewModel.numberOfLoadingTasks++
		})

		$rootScope.$on("remove-loading-task", () => {
			this.viewModel.numberOfLoadingTasks--
		})
	}
}

export const codeChartaComponent = {
	selector: "codeChartaComponent",
	template: require("./codeCharta.component.html"),
	controller: CodeChartaController
}
