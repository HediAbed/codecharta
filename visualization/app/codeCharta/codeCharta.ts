"use strict";

import "./core/core.module";
import "./ui/ui";

import {codeChartaComponent} from "./codeCharta.component";

import angular from "angular";
import { CodeChartaService } from "./codeCharta.service";
import _ from "lodash";

angular.module(
    "app.codeCharta",
    ["app.codeCharta.core", "app.codeCharta.ui"]
);

angular.module("app.codeCharta").component(
    codeChartaComponent.selector,
    codeChartaComponent
).service(
    _.camelCase(CodeChartaService.name),
    CodeChartaService
);


