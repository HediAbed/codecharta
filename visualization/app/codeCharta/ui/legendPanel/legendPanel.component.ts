import $ from "jquery"
import { IRootScopeService } from "angular"
import "./legendPanel.component.scss"
import { ColorRange, MarkedPackage } from "../../codeCharta.model"
import { CodeChartaService } from "../../codeCharta.service"
import { FileStateService } from "../../state/fileState.service"
import { FileStateHelper } from "../../util/fileStateHelper"
import { ColorConverter } from "../../util/color/colorConverter"
import { AttributeSideBarService, AttributeSideBarVisibilitySubscriber } from "../attributeSideBar/attributeSideBar.service"
import { ColorRangeService, ColorRangeSubscriber } from "../../state/store/dynamicSettings/colorRange/colorRange.service"
import {
	InvertColorRangeService,
	InvertColorRangeSubscriber
} from "../../state/store/appSettings/invertColorRange/invertColorRange.service"
import { StoreService } from "../../state/store.service"
import { MarkedPackagesSubscriber } from "../../state/store/fileSettings/markedPackages/markedPackages.service"

export interface PackageList {
	colorPixel: string
	markedPackages: MarkedPackage[]
}

export class LegendPanelController
	implements AttributeSideBarVisibilitySubscriber, ColorRangeSubscriber, InvertColorRangeSubscriber, MarkedPackagesSubscriber {
	private _viewModel: {
		isLegendVisible: boolean
		isSideBarVisible: boolean
		isDeltaState: boolean
		colorRange: ColorRange
		invertColorRange: boolean
		packageLists: PackageList[]
	} = {
		isLegendVisible: false,
		isSideBarVisible: null,
		isDeltaState: null,
		colorRange: null,
		invertColorRange: null,
		packageLists: null
	}

	/* @ngInject */
	constructor(private $rootScope: IRootScopeService, private storeService: StoreService, private fileStateService: FileStateService) {
		ColorRangeService.subscribe(this.$rootScope, this)
		InvertColorRangeService.subscribe(this.$rootScope, this)
		AttributeSideBarService.subscribe(this.$rootScope, this)
	}

	public onColorRangeChanged(colorRange: ColorRange) {
		this._viewModel.colorRange = colorRange
		this.updatePixelColors()
	}

	public onInvertColorRangeChanged(invertColorRange: boolean) {
		this._viewModel.invertColorRange = invertColorRange
		this.updatePixelColors()
	}

	public onMarkedPackagesChanged(markedPackages: MarkedPackage[]) {
		this.setMarkedPackageLists(markedPackages)
	}

	public onAttributeSideBarVisibilityChanged(isAttributeSideBarVisible: boolean) {
		this._viewModel.isSideBarVisible = isAttributeSideBarVisible
	}

	public toggle() {
		this._viewModel.isLegendVisible = !this._viewModel.isLegendVisible
	}

	private updatePixelColors() {
		this._viewModel.isDeltaState = FileStateHelper.isDeltaState(this.fileStateService.getFileStates())
		this.setPixel("selected", this.storeService.getState().appSettings.mapColors.selected)

		if (this._viewModel.isDeltaState) {
			this.updateDeltaColors()
		} else {
			this.updateNormalColors()
		}
	}

	private updateNormalColors() {
		const mapColors = this.storeService.getState().appSettings.mapColors
		const positive = this.storeService.getState().appSettings.whiteColorBuildings ? mapColors.lightGrey : mapColors.positive

		this.setPixel("positive", positive)
		this.setPixel("neutral", mapColors.neutral)
		this.setPixel("negative", mapColors.negative)
		this.setPixel("incomingEdge", mapColors.incomingEdge)
		this.setPixel("outgoingEdge", mapColors.outgoingEdge)
	}

	private updateDeltaColors() {
		const mapColors = this.storeService.getState().appSettings.mapColors
		const positiveDelta = this.storeService.getState().appSettings.invertDeltaColors ? mapColors.negativeDelta : mapColors.positiveDelta
		const negativeDelta = this.storeService.getState().appSettings.invertDeltaColors ? mapColors.positiveDelta : mapColors.negativeDelta

		this.setPixel("positiveDelta", positiveDelta)
		this.setPixel("negativeDelta", negativeDelta)
	}

	private setPixel(id: string, color: string) {
		$("#" + id).attr("src", ColorConverter.getImageDataUri(color))
	}

	private setMarkedPackageLists(markedPackages: MarkedPackage[]) {
		if (markedPackages) {
			this._viewModel.packageLists = []
			markedPackages.forEach(mp => this.handleMarkedPackage(mp))
		}
	}

	private handleMarkedPackage(mp: MarkedPackage) {
		const colorPixel = ColorConverter.getImageDataUri(mp.color)

		if (!mp.attributes["name"]) {
			mp.attributes["name"] = this.getPackagePathPreview(mp)
		}

		if (this.isColorPixelInPackageLists(colorPixel)) {
			this.insertMPIntoPackageList(mp, colorPixel)
		} else {
			const packageList: PackageList = { colorPixel: colorPixel, markedPackages: [mp] }
			this.addNewPackageList(packageList)
		}
	}

	private addNewPackageList(packageList: PackageList) {
		this._viewModel.packageLists.push(packageList)
	}

	private insertMPIntoPackageList(mp: MarkedPackage, colorPixel: string) {
		this._viewModel.packageLists.filter(packageList => packageList.colorPixel == colorPixel)[0].markedPackages.push(mp)
	}

	private isColorPixelInPackageLists(colorPixel: string) {
		return this._viewModel.packageLists.filter(mpList => mpList.colorPixel == colorPixel).length > 0
	}

	private getPackagePathPreview(mp: MarkedPackage): string {
		const MAX_NAME_LENGTH = { lowerLimit: 24, upperLimit: 28 }
		const packageName = this.getPackageNameFromPath(mp.path)

		if (packageName.length > MAX_NAME_LENGTH.lowerLimit && packageName.length < MAX_NAME_LENGTH.upperLimit) {
			return ".../" + packageName
		} else if (packageName.length > MAX_NAME_LENGTH.upperLimit) {
			const firstPart = packageName.substr(0, MAX_NAME_LENGTH.lowerLimit / 2)
			const secondPart = packageName.substr(packageName.length - MAX_NAME_LENGTH.lowerLimit / 2)
			return firstPart + "..." + secondPart
		} else {
			const from = Math.max(mp.path.length - MAX_NAME_LENGTH.lowerLimit, 0)
			const previewPackagePath = mp.path.substring(from)
			const startingDots = previewPackagePath.startsWith(CodeChartaService.ROOT_PATH) ? "" : "..."
			return startingDots + previewPackagePath
		}
	}

	private getPackageNameFromPath(path: string): string {
		const pathArray = path.split("/")
		return pathArray[pathArray.length - 1]
	}
}

export const legendPanelComponent = {
	selector: "legendPanelComponent",
	template: require("./legendPanel.component.html"),
	controller: LegendPanelController
}
