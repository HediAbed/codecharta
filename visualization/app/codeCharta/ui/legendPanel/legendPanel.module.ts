import angular from "angular"
import "../../state/state.module"
import "../codeMap/codeMap.module"
import "../../codeCharta.module"

import { legendPanelComponent } from "./legendPanel.component"

angular
	.module("app.codeCharta.ui.legendPanel", ["app.codeCharta.state", "app.codeCharta.ui.codeMap", "app.codeCharta"])
	.component(legendPanelComponent.selector, legendPanelComponent)
