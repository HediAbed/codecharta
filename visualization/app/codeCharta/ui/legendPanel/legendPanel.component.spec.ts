import "./legendPanel.module"

import { LegendPanelController, PackageList } from "./legendPanel.component"
import { SettingsService } from "../../state/settings.service"
import { instantiateModule, getService } from "../../../../mocks/ng.mockhelper"
import { CodeChartaService } from "../../codeCharta.service"
import { FileStateService } from "../../state/fileState.service"
import { IRootScopeService } from "angular"

describe("LegendPanelController", () => {
	let services, legendPanelController: LegendPanelController

	beforeEach(() => {
		restartSystem()
		rebuildController()
	})

	function restartSystem() {
		instantiateModule("app.codeCharta.ui.legendPanel")

		services = {
			$rootScope: getService<IRootScopeService>("$rootScope"),
			settingsService: getService<SettingsService>("settingsService"),
			codeChartaService: getService<CodeChartaService>("codeChartaService"),
			fileStateService: getService<FileStateService>("fileStateService"),
		}
	}

	function rebuildController() {
		legendPanelController = new LegendPanelController(
			services.$rootScope,
			services.settingsService,
			services.codeChartaService,
			services.fileStateService
		)
	}

	describe("MarkingColor in Legend", () => {
		beforeEach(() => {
			let simpleHierarchy = {
				name: "root",
				type: "Folder",
				path: "/root",
				attributes: {},
				children: [
					{
						name: "a",
						type: "Folder",
						path: "/root/a",
						attributes: {},
						children: [
							{
								name: "aa",
								type: "File",
								path: "/root/a/aa",
								attributes: {}
							},
							{
								name: "ab",
								type: "Folder",
								path: "/root/a/ab",
								attributes: {},
								children: [
									{
										name: "aba",
										path: "/root/a/ab/aba",
										type: "File",
										attributes: {}
									}
								]
							}
						]
					},
					{
						name: "b",
						type: "File",
						path: "/root/b",
						attributes: {}
					}
				]
			}

			services.fileStateService.fileStates.push({ file: simpleHierarchy, selectedAs: null })
		})

		it("set correct markingPackage in Legend", () => {
			services.settingsService.settings.fileSettings.markedPackages = [{ color: "#FF0000", path: "/root", attributes: {} }]
			const expectedPackageLists: PackageList[] = [
				{
					colorPixel: "data:image/gif;base64,R0lGODlhAQABAPAAAP8AAP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
					markedPackages: [{ color: "#FF0000", path: "/root", attributes: { name: "/root" } }]
				}
			]

			legendPanelController.onSettingsChanged(services.settingsService.settings, null)

			expect(legendPanelController["_viewModel"].packageLists).toEqual(expectedPackageLists)
		})

		it("shorten too long pathName in middle of the string for legendPanel", () => {
			services.settingsService.settings.fileSettings.markedPackages = [
				{ color: "#FF0000", path: "/root/a/longNameToBeShortenedInLegend", attributes: {} }
			]
			const shortenedPathname = "longNameToBe...enedInLegend"

			legendPanelController.onSettingsChanged(services.settingsService.settings, null)
			expect(legendPanelController["_viewModel"].packageLists[0].markedPackages[0].attributes["name"]).toEqual(shortenedPathname)
		})

		it("shorten too long pathName at beginning of the string for legendPanel", () => {
			services.settingsService.settings.fileSettings.markedPackages = [
				{ color: "#FF0000", path: "/root/a/andAnotherLongNameToShorten", attributes: {} }
			]
			const shortenedPathname = ".../andAnotherLongNameToShorten"

			legendPanelController.onSettingsChanged(services.settingsService.settings, null)
			expect(legendPanelController["_viewModel"].packageLists[0].markedPackages[0].attributes["name"]).toEqual(shortenedPathname)
		})
	})
})
