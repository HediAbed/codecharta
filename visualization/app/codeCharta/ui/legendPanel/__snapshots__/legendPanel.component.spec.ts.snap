// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendPanelController onColorRangeChanged should update the color range and pixels in viewModel when it is changed 1`] = `
Object {
  "incomingEdge": "data:image/gif;base64,R0lGODlhAQABAPAAAAD//////yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "negative": "data:image/gif;base64,R0lGODlhAQABAPAAAIIODv///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "neutral": "data:image/gif;base64,R0lGODlhAQABAPAAAN3MAP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "outgoingEdge": "data:image/gif;base64,R0lGODlhAQABAPAAAP8A/////yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "positive": "data:image/gif;base64,R0lGODlhAQABAPAAAGmuQP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "selected": "data:image/gif;base64,R0lGODlhAQABAPAAAOuDGf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
}
`;

exports[`LegendPanelController onInvertColorRangeChanged should update the invertColorRange and pixels in viewModel when it is changed 1`] = `
Object {
  "incomingEdge": "data:image/gif;base64,R0lGODlhAQABAPAAAAD//////yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "negative": "data:image/gif;base64,R0lGODlhAQABAPAAAIIODv///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "neutral": "data:image/gif;base64,R0lGODlhAQABAPAAAN3MAP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "outgoingEdge": "data:image/gif;base64,R0lGODlhAQABAPAAAP8A/////yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "positive": "data:image/gif;base64,R0lGODlhAQABAPAAAGmuQP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "selected": "data:image/gif;base64,R0lGODlhAQABAPAAAOuDGf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
}
`;

exports[`LegendPanelController onInvertDeltaColors should update the pixels in viewModel when it is changed 1`] = `
Object {
  "incomingEdge": "data:image/gif;base64,R0lGODlhAQABAPAAAAD//////yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "negative": "data:image/gif;base64,R0lGODlhAQABAPAAAIIODv///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "neutral": "data:image/gif;base64,R0lGODlhAQABAPAAAN3MAP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "outgoingEdge": "data:image/gif;base64,R0lGODlhAQABAPAAAP8A/////yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "positive": "data:image/gif;base64,R0lGODlhAQABAPAAAGmuQP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "selected": "data:image/gif;base64,R0lGODlhAQABAPAAAOuDGf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
}
`;

exports[`LegendPanelController onWhiteColorBuildings should update the pixels in viewModel when it is changed 1`] = `
Object {
  "incomingEdge": "data:image/gif;base64,R0lGODlhAQABAPAAAAD//////yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "negative": "data:image/gif;base64,R0lGODlhAQABAPAAAIIODv///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "neutral": "data:image/gif;base64,R0lGODlhAQABAPAAAN3MAP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "outgoingEdge": "data:image/gif;base64,R0lGODlhAQABAPAAAP8A/////yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "positive": "data:image/gif;base64,R0lGODlhAQABAPAAAGmuQP///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
  "selected": "data:image/gif;base64,R0lGODlhAQABAPAAAOuDGf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==",
}
`;
